src/Microsoft.PowerShell.Commands.Utility/commands/utility/select-object.cs

```diff
@@ -337,6 +337,11 @@ private void ProcessExpressionParameter()
             if (ExcludeProperty != null)
             {
                 _exclusionFilter = new MshExpressionFilter(ExcludeProperty);
+                if ((Property == null) || (Property.Length == 0))
+                {
```

Comment:
Sorry @iSazonov for the long delay.
@lzybkr and I had a quick discussion about this, and I think @lzybkr agreed that we should take the original fix with the *unlikely* breaking change.